--- pu-a/Makefile	2026-09-02 12:31:44.724052748 +0800
+++ pu-b/Makefile	2026-09-02 12:31:44.726633779 +0800
@@ -16,6 +16,7 @@
 # OUTDIR is not used with uefi/*.o deliberately
 OUTDIR:=$(addsuffix /,$(OUTDIR))
 endif
+CFLAGS += -DUEFI_NO_UTF8
 CFLAGS += -fshort-wchar -fno-strict-aliasing -ffreestanding -fno-stack-protector -fno-stack-check -I. -I./uefi \
   -I/usr/include -I/usr/include/efi -I/usr/include/efi/protocol -I/usr/include/efi/$(ARCH) -D__$(ARCH)__
 ifeq ($(ARCH),x86_64)
@@ -83,7 +84,7 @@
 endif
 
 uefi/libuefi.a:
-	@$(MAKE) --no-print-directory -C uefi libuefi.a USE_GCC=$(USE_GCC) ARCH=$(ARCH)
+	@$(MAKE) --no-print-directory -C uefi libuefi.a USE_GCC=$(USE_GCC) ARCH=$(ARCH) OUTDIR=
 
 libuefi.lib: $(LIBOBJS)
 
--- pu-a/uefi/Makefile	2026-02-01 11:39:21.926472189 +0800
+++ pu-b/uefi/Makefile	2026-09-02 12:31:44.727459116 +0800
@@ -16,6 +16,7 @@
 # OUTDIR is not used with uefi/*.o deliberately
 OUTDIR:=$(addsuffix /,$(OUTDIR))
 endif
+CFLAGS += -DUEFI_NO_UTF8
 CFLAGS += -fshort-wchar -fno-strict-aliasing -ffreestanding -fno-stack-protector -fno-stack-check -I. -I./uefi \
   -I/usr/include -I/usr/include/efi -I/usr/include/efi/protocol -I/usr/include/efi/$(ARCH) -D__$(ARCH)__
 ifeq ($(ARCH),x86_64)
